License identified in scripts/Makefile.build:
# SPDX-License-Identifier: GPL-2.0

These changes are for dkms to reduce the length of abspaths generated for each
.o file before linking.

--- a/scripts/Makefile.build.orig	2023-11-18 16:24:11.345131407 -0800
+++ b/scripts/Makefile.build	2023-11-18 17:00:42.785074727 -0800
@@ -428,9 +428,11 @@ endif
 # module is turned into a multi object module, $^ will contain header file
 # dependencies recorded in the .*.cmd file.
 quiet_cmd_link_multi-m = LD [M]  $@
-      cmd_link_multi-m = $(LD) $(ld_flags) -r -o $@ $(filter %.o,$^)
+      cmd_link_multi-m = $(LD) $(strip $(ld_flags)) -r -o $@ @$@.in
 
 $(multi-used-m): FORCE
+	$(file >$@.in)
+	$(foreach O,$(filter %.o,$^),$(file >>$(realpath $@.in),$O))
 	$(call if_changed,link_multi-m)
 $(call multi_depend, $(multi-used-m), .o, -objs -y -m)
 
